fe/src/main/java/org/apache/doris/analysis/Expr.java

```diff
@@ -1462,6 +1462,18 @@ public Subquery getSubquery() {
         return subqueries.get(0);
     }
 
+    public boolean isCorrelatedPredicate(List<TupleId> tupleIdList) {
+        if ((this instanceof BinaryPredicate || this instanceof SlotRef) && !this.isBoundByTupleIds(tupleIdList)) {
+            return true;
+        }
```

Comment:
why you add “this instanceof BinaryPredicate ”？
only slotref has override isBoundByTupleIds function.